integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py

```diff
@@ -1537,26 +1535,31 @@ def _prepare_collection_params(self):
             "init_from": self.init_from,
         }
 
-    def _prepare_client_params(self):
+    def _prepare_client_params(self) -> Dict[str, Any]:
         """
         Prepares the common parameters for client initialization.
 
         """
-        return {
-            "location": self.location,
-            "url": self.url,
-            "port": self.port,
-            "grpc_port": self.grpc_port,
-            "prefer_grpc": self.prefer_grpc,
-            "https": self.https,
-            "api_key": self.api_key.resolve_value() if self.api_key else None,
-            "prefix": self.prefix,
-            "timeout": self.timeout,
-            "host": self.host,
-            "path": self.path,
-            "metadata": self.metadata,
-            "force_disable_check_same_thread": self.force_disable_check_same_thread,
-        }
+        # NOTE: We need to use deepcopy here to avoid modifying the original class attributes.
+        # For example, the resolved api key is added to metadata by the QdrantClient class when using a hosted
+        # Qdrant service, which means running to_dict() exposes the api key.
+        return deepcopy(
+            {
+                "location": self.location,
+                "url": self.url,
+                "port": self.port,
+                "grpc_port": self.grpc_port,
+                "prefer_grpc": self.prefer_grpc,
+                "https": self.https,
+                "api_key": self.api_key.resolve_value() if self.api_key else None,
+                "prefix": self.prefix,
+                "timeout": self.timeout,
+                "host": self.host,
+                "path": self.path,
+                "metadata": self.metadata,
```

Comment:
Wouldn't it be enough to only deepcopy metadata? Or maybe something like: `"metadata": {**self.metadata}`?

All other params are primitives and they should not be affected when the QDrantClient changes them. Or am I missing something?

Reply:
yeah that's fair we could just target the metadata field in this case